plugins/poll/app/controllers/polls_controller.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+class DiscoursePoll::PollsController < ::ApplicationController
+  requires_plugin DiscoursePoll::PLUGIN_NAME
+
+  before_action :ensure_logged_in, except: [:voters, :grouped_poll_results]
+
+  def vote
+    post_id = params.require(:post_id)
+    poll_name = params.require(:poll_name)
+    options = params.require(:options)
+
+    begin
+      poll, options = DiscoursePoll::Poll.vote(post_id, poll_name, options, current_user)
+      render json: { poll: poll, vote: options }
+    rescue DiscoursePoll::Error => e
+      render_json_error e.message
+    end
+  end
+
+  def remove_vote
+    post_id = params.require(:post_id)
+    poll_name = params.require(:poll_name)
+
+    begin
+      poll = DiscoursePoll::Poll.remove_vote(post_id, poll_name, current_user)
+      render json: { poll: poll }
+    rescue DiscoursePoll::Error => e
+      render_json_error e.message
+    end
+  end
+
+  def toggle_status
+    post_id = params.require(:post_id)
+    poll_name = params.require(:poll_name)
+    status = params.require(:status)
+
+    begin
+      poll = DiscoursePoll::Poll.toggle_status(post_id, poll_name, status, current_user)
+      render json: { poll: poll }
+    rescue DiscoursePoll::Error => e
+      render_json_error e.message
+    end
+  end
+
+  def voters
+    post_id = params.require(:post_id)
+    poll_name = params.require(:poll_name)
+    opts = params.permit(:limit, :page, :option_id)
+
+    post = Post.find_by(id: post_id)
```

Comment:
Since you're not using `post` anywhere else, there's no point in loading it using `find_by`. Maybe just use `exists?`